tests/pipeline/test_multidimensional_dataset.py

```diff
@@ -60,13 +60,38 @@ class MD2(DataSetFamily):
         assert_is_not(MD1Slice, MD2Slice)
 
     def test_empty_extra_dims(self):
-        expected_msg = (
-            'DataSetFamily must be defined with non-empty extra_dims'
+        msg = (
+            "^DataSetFamily must be defined with non-empty extra_dims,"
+            " or with `__abstract__ = True`$"
         )
-        with assert_raises_str(ValueError, expected_msg):
-            class MD(DataSetFamily):
+        with self.assertRaisesRegexp(ValueError, msg):
```

Comment:
why the change to a regex with no pattern?

Reply:
Didn't see that we had an `assert_raises_str`. I'll use that instead. Thanks!